cdap-docs/developer-manual/source/metadata/programmatic-metadata.rst

```diff
@@ -0,0 +1,91 @@
+.. meta::
+    :author: Cask Data, Inc.
+    :copyright: Copyright © 2018 Cask Data, Inc.
+
+.. _programmatic-metadata:
+
+==================================
+Metadata from Programs and Plugins
+==================================
+
+.. highlight:: java
+
+Metadata can be accessed from *Programs* and *Plugins*. Currently, reading metadata from a program or plugin is not
+supported while it is running in *cloud* but metadata can be added, updated or deleted.
+
+.. _metadata-programs:
+
+Programs
+========
+
+Metadata can be accessed from *MapReduce*, *Spark*, *Workers* and *Service* through
```

Comment:
... through methods from the MetadataReader object, which are available via the appropriate program context object in your program. The program context object can be obtained by invoking the ``getContext()`` method inside the ``initialize`` method of your program.

Reply:
Done.